### PR TITLE
rpm,deb,qa,python-common,test: drop python2 support

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
jenkins test dashboard backend

**Comment by tchaikov** [comment]
@sebastian-philipp why is this change related to ceph-volume?

**Comment by sebastian-philipp** [comment]
> @sebastian-philipp why is this change related to ceph-volume?

Cause of #35728

File: src/pybind/mgr/rook/rook_cluster.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 from time import sleep
 
 import jsonpatch
-from six.moves.urllib.parse import urljoin  # pylint: disable=import-error
+from urllib.parse import urljoin  # pylint: disable=import-error
```
Comment (sebastian-philipp) [inline src/pybind/mgr/rook/rook_cluster.py:17]:
```suggestion
from urllib.parse import urljoin
```

Comment (tchaikov) [reply src/pybind/mgr/rook/rook_cluster.py:17]: removed and repushed.

File: qa/tasks/mgr/dashboard/helper.py [hunk 2]
```diff
@@ -462,7 +461,7 @@ def find_object_in_list(cls, key, value, iterable):
 class JLeaf(namedtuple('JLeaf', ['typ', 'none'])):
     def __new__(cls, typ, none=False):
         if typ == str:
-            typ = six.string_types
+            typ = str
```
Comment (dillaman) [inline qa/tasks/mgr/dashboard/helper.py:464]: Nit: unnecessary?

Comment (tchaikov) [reply qa/tasks/mgr/dashboard/helper.py:464]: updated and repushed.

**Comment by jdurgin** [comment]
retest this please

**Comment by tchaikov** [comment]
> > @sebastian-philipp why is this change related to ceph-volume?
> 
> This PR is going to make it hard to port #35728 PR to Nautilus. Which is not the goal anyway, but I still would like to get @jan--f 's OK

@sebastian-philipp  @jan--f  i can help with the backport of #35728, if the change in python-common concerns you.

**Comment by callithea** [comment]
jenkins test dashboard backend

**Comment by callithea** [comment]
jenkins test dashboard backend

**Comment by tchaikov** [comment]
changelog

- fix `JLeaf` by adding `defaults`.

**Comment by batrick** [comment]
https://pulpito.ceph.com/?branch=wip-pdonnell-testing-20200721.040813

CephFS failures look unrelated. @tchaikov let me know if you want me to retest with your latest changes.

**Comment by neha-ojha** [comment]
retest this please

**Comment by tchaikov** [comment]
> https://pulpito.ceph.com/?branch=wip-pdonnell-testing-20200721.040813
> 
> CephFS failures look unrelated. @tchaikov let me know if you want me to retest with your latest changes.

thanks @batrick for testing this changeset, no need to retest them. AFACT, `JLeaf` is only used by the mgr dashboard backend to validate the schema of mgr's interfaces for retrieving cluster information. i will take care of the dashboard tests when testing rados suites.

**Comment by tchaikov** [comment]
changelog

- fix `JLeaf` by using `__new__`. `defaults` was added in python 3.7, while we are using python3.6 when running "make check".

**Comment by callithea** [comment]
jenkins test dashboard

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2020-07-21_07:32:05-rados-wip-kefu-testing-2020-07-21-0014-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-07-22_03:46:42-rados-wip-kefu-testing-2020-07-21-2148-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-07-22_06:32:16-rados-wip-kefu-testing-2020-07-22-1235-distro-basic-smithi/

**Comment by tchaikov** [comment]
@callithea yes, but it was fixed in the latest revision. the failure was caused by the buggy change to `JLeaf`.

**Pull request merged**